docs/src/architecture/accounts.md

```diff
@@ -13,63 +16,56 @@ In the above:
 
-* Accounts with **public state**, where the actual state is stored on chain. These would be similar to how accounts work in public blockchains.
-* Accounts with **private state**, where only the hash of the account is stored on chain. The hash is defined as: \
-`hash([account ID, 0, 0, nonce], [vault root], [storage root], [code root])`
+| | Regular updatable account | Regular immutable account | Faucet for fungible assets | Faucet for non-fungible assets |
+|---|---|---|---|---|
+| **Description** | Will be used by most users for a wallet. Code specification and changes possible. | Will be used by most smart contracts. Once deployed code cannot be changed | Users can issue fungible assets and customize them. | Users can issue non-fungible assets and customize them. |
+| **Code updatability** | yes | no | no | no |
+| **Most significant bits** | `00` | `01` | `10` | `11` |
 
-### Example account ID
+#### Example account ID (big-endian)
 
 <p align="center">
     <img src="../diagrams/architecture/account/Account_ID.png">
 </p>
 
-## Storage
-User-defined data that can be stored in an account. Account storage is composed of two components. The first component is a simple sparse Merkle tree of depth 8 which is index addressable. This provides the user with 256 Word slots. Users that require additional storage can use the second component which is a `MerkleStore`. This will allow the user to store any Merkle structures they need. This is achieved by storing the root of the Merkle structure as a leaf in the simple sparse merkle tree. When `AccountStorage` is serialized it will check to see if any of the leafs in the simple sparse Merkle tree are Merkle roots of other Merkle structures. If any Merkle roots are found then the Merkle structures will be persisted in the `AccountStorage` `MerkleStore`.
+*Note: Miden uses little-endian by default, but big_endian for better readability in the picture above.*
 
-TODO ADD STORAGE DIAGRAM
+### Account Storage
+User-defined data that can be stored in an account. `AccountStorage` is composed of two components. 
 
-## Nonce
+The first component is a simple sparse Merkle tree of depth `8` which is index addressable. This provides the user with `256` `Word` slots. 
+
+Users requiring additional storage can use the second component a `MerkleStore`. It allows users to store any Merkle structures they need. The root of the Merkle structure can be stored as a leaf in the simple sparse Merkle tree. When `AccountStorage` is serialized it will check to see if any of the leafs in the simple sparse Merkle tree are Merkle roots of other Merkle structures. If any Merkle roots are found then the Merkle structures will be persisted in the `AccountStorage` `MerkleStore`.
```

Comment:
Users requiring additional storage can use the second component a `MerkleStore`, which allows users to store any Merkle structures they need.

Reply:
Is there a difference between your suggestion using a relative clause and mine using two sentences? I am not a native speaker

Reply:
I just think it read better but not strongly opinionated.